src/librustc_mir/interpret/validity.rs

```diff
@@ -715,7 +714,8 @@ impl<'rt, 'mir, 'tcx, M: Machine<'mir, 'tcx>> ValueVisitor<'mir, 'tcx, M>
         fields: impl Iterator<Item = InterpResult<'tcx, Self::V>>,
     ) -> InterpResult<'tcx> {
         match op.layout.ty.kind {
-            ty::Str => {
+            // FIXME(eddyb) does this belong here?
```

Comment:
Well you could remove that branch and then str would not be special any more in terms of validity. ;)